### ISSUE TITLE
Error `no method matching find_installed` on updating single package with Julia v1.2-rc1

### ISSUE THREAD

**Issue by oschulz** [JuliaLang/Pkg.jl#1216]
Just tried to update a single package, and got this 

```
(v1.2) pkg> update TypedTables
  Updating registry at `/user/.julia/registries/General`
  Updating git-repo `https://github.com/JuliaRegistries/General.git`
 Resolving package versions...
ERROR: MethodError: no method matching find_installed(::String, ::Base.UUID, ::Nothing)
Closest candidates are:
  find_installed(::String, ::Base.UUID, ::Base.SHA1) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.2/Pkg/src/Operations.jl:20
Stacktrace:
 [1] source_path(::Pkg.Types.PackageSpec) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.2/Pkg/src/Operations.jl:37
 [2] collect_fixed!(::Pkg.Types.Context, ::Array{Pkg.Types.PackageSpec,1}, ::Dict{Base.UUID,String}) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.2/Pkg/src/Operations.jl:246
 [3] resolve_versions!(::Pkg.Types.Context, ::Array{Pkg.Types.PackageSpec,1}) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.2/Pkg/src/Operations.jl:291
 [4] up(::Pkg.Types.Context, ::Array{Pkg.Types.PackageSpec,1}, ::Pkg.Types.UpgradeLevel) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.2/Pkg/src/Operations.jl:1051
 [5] #up#43(::Pkg.Types.UpgradeLevel, ::Pkg.Types.PackageMode, ::Bool, ::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}, ::typeof(Pkg.API.up), ::Pkg.Types.Context, ::Array{Pkg.Types.PackageSpec,1}) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.2/Pkg/src/API.jl:167
 [6] up at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.2/Pkg/src/API.jl:143 [inlined]
 [7] do_up!(::Dict{Symbol,Any}, ::Array{Pkg.Types.PackageSpec,1}, ::Dict{Symbol,Any}) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.2/Pkg/src/REPLMode.jl:483
 [8] #invokelatest#1(::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}, ::typeof(Base.invokelatest), ::Any, ::Any, ::Vararg{Any,N} where N) at ./essentials.jl:790
 [9] invokelatest(::Any, ::Any, ::Vararg{Any,N} where N) at ./essentials.jl:789
 [10] do_cmd!(::Pkg.REPLMode.Command, ::REPL.LineEditREPL) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.2/Pkg/src/REPLMode.jl:403


julia> versioninfo()
Julia Version 1.2.0-rc1.0
Commit 7097799cf1 (2019-05-30 02:22 UTC)
Platform Info:
  OS: Linux (x86_64-pc-linux-gnu)
  CPU: Intel(R) Core(TM) i7-7500U CPU @ 2.70GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-6.0.1 (ORCJIT, skylake)
Environment:
  JULIA_DEPOT_PATH = /user/.julia
  JULIA_EDITOR = sublime_text_venv
  JULIA_ARRAYFIRE_BACKEND = cpu
  JULIA_PKGDIR = /user/.julia
  JULIA_PKG_DEVDIR = /user/.julia/dev
  JULIA_CXX_RTTI = 1
```

**Comment by fredrikekre** [comment]
toml files?

**Comment by oschulz** [comment]
Ooops, yes, sorry. Here's my v1.2 Project.toml and Manifest.toml:

 https://gist.github.com/oschulz/6c593f2526558c88c402af4ac91fcc73

Haven't managed to reproduce this with a fresh Project.toml and Manifest.toml yet.

**Comment by fredrikekre** [comment]
172 deps :scream: 

Anyway, the toml files include a bunch of local file paths so difficult to debug them. However, I think this is the problem: https://gist.github.com/oschulz/6c593f2526558c88c402af4ac91fcc73#file-manifest-toml-L138-L145

Would be good to figure out why this errors though. Did you manually edit the Manifest.toml file?

**Comment by oschulz** [comment]
> 172 deps

I know ... sorry, it's my default package list ... I tried to make a shorter one that reproduces the error, but haven't been successful so far.

No the `Manifest.toml` has never been changed by hand.

**Comment by fredrikekre** [comment]
I removed everything I thought was not important, but can't reproduce the error. Can you try:
```
using Pkg
@eval Pkg.Operations begin
    function source_path(pkg::PackageSpec)
        @show pkg.name, pkg.uuid, pkg.path, pkg.repo.url, pkg.tree_hash
        return is_stdlib(pkg.uuid)    ? Types.stdlib_path(pkg.name) :
            pkg.path      !== nothing ? pkg.path :
            pkg.repo.url  !== nothing ? find_installed(pkg.name, pkg.uuid, pkg.tree_hash) :
            pkg.tree_hash !== nothing ? find_installed(pkg.name, pkg.uuid, pkg.tree_hash) :
            nothing
    end
end
```
and show the output from Pkg.add that errors?

**Comment by oschulz** [comment]
Thanks, here goes:

```
julia> @eval Pkg.Operations begin
           function source_path(pkg::PackageSpec)
               @show pkg.name, pkg.uuid, pkg.path, pkg.repo.url, pkg.tree_hash
               return is_stdlib(pkg.uuid)    ? Types.stdlib_path(pkg.name) :
                   pkg.path      !== nothing ? pkg.path :
                   pkg.repo.url  !== nothing ? find_installed(pkg.name, pkg.uuid, pkg.tree_hash) :
                   pkg.tree_hash !== nothing ? find_installed(pkg.name, pkg.uuid, pkg.tree_hash) :
                   nothing
           end
       end
source_path (generic function with 1 method)

(v1.2) pkg> update TypedTables
  Updating registry at `/user/.julia/registries/General`
  Updating git-repo `https://github.com/JuliaRegistries/General.git`
 Resolving package versions...
(pkg.name, pkg.uuid, pkg.path, pkg.repo.url, pkg.tree_hash) = ("TypedTables", UUID("9d95f2ec-7b3d-5a63-8d20-e2491e220bb9"), "/user/.julia/dev/TypedTables", nothing, nothing)
(pkg.name, pkg.uuid, pkg.path, pkg.repo.url, pkg.tree_hash) = ("MyJuliaPackage", UUID("e5f32138-46ba-5297-a00b-70d7706f31cf"), "/user/.julia/dev/MyJuliaPackage", nothing, nothing)
(pkg.name, pkg.uuid, pkg.path, pkg.repo.url, pkg.tree_hash) = ("LegendUpROOTIO", UUID("59dd4b32-a3c2-596d-b22d-5bb8b5f60033"), "/user/.julia/dev/LegendUpROOTIO", nothing, nothing)
(pkg.name, pkg.uuid, pkg.path, pkg.repo.url, pkg.tree_hash) = ("UnsafeArrays", UUID("c4a57d5a-5b31-53a6-b365-19f8c011fbd6"), "/user/.julia/dev/UnsafeArrays", nothing, nothing)
(pkg.name, pkg.uuid, pkg.path, pkg.repo.url, pkg.tree_hash) = ("RecipesBase", UUID("3cdcf5f2-1ef4-517c-9805-6587b60abb01"), "/user/.julia/dev/RecipesBase", nothing, nothing)
(pkg.name, pkg.uuid, pkg.path, pkg.repo.url, pkg.tree_hash) = ("MJDSigGen", UUID("afe346ea-f9a0-5cb0-94b3-bc7b131cb9d1"), "/user/.julia/dev/MJDSigGen", nothing, nothing)
(pkg.name, pkg.uuid, pkg.path, pkg.repo.url, pkg.tree_hash) = ("BoostFractor", UUID("222d2961-0b53-58be-99ce-9613a82efb32"), "/user/.julia/dev/BoostFractor", nothing, nothing)
(pkg.name, pkg.uuid, pkg.path, pkg.repo.url, pkg.tree_hash) = ("AbstractPlotting", UUID("537997a7-5e4e-5d89-9595-2241ea00577e"), "/user/.julia/dev/AbstractPlotting", nothing, nothing)
(pkg.name, pkg.uuid, pkg.path, pkg.repo.url, pkg.tree_hash) = ("LegendDSP", UUID("e6d7a502-be5c-5049-a927-8d60e566555d"), "/user/.julia/dev/LegendDSP", nothing, nothing)
(pkg.name, pkg.uuid, pkg.path, pkg.repo.url, pkg.tree_hash) = ("SixelTerm", UUID("91ba51fb-686c-5ec1-90fe-e45fdc68a5b2"), nothing, "https://github.com/tshort/SixelTerm.jl.git", nothing)
ERROR: MethodError: no method matching find_installed(::String, ::Base.UUID, ::Nothing)
Closest candidates are:
  find_installed(::String, ::Base.UUID, ::Base.SHA1) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.2/Pkg/src/Operations.jl:20
Stacktrace:
 [1] source_path(::Pkg.Types.PackageSpec) at ./REPL[2]:4
 [2] collect_fixed!(::Pkg.Types.Context, ::Array{Pkg.Types.PackageSpec,1}, ::Dict{Base.UUID,String}) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.2/Pkg/src/Operations.jl:246
 [3] resolve_versions!(::Pkg.Types.Context, ::Array{Pkg.Types.PackageSpec,1}) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.2/Pkg/src/Operations.jl:291
 [4] up(::Pkg.Types.Context, ::Array{Pkg.Types.PackageSpec,1}, ::Pkg.Types.UpgradeLevel) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.2/Pkg/src/Operations.jl:1051
 [5] #up#43(::Pkg.Types.UpgradeLevel, ::Pkg.Types.PackageMode, ::Bool, ::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}, ::typeof(Pkg.API.up), ::Pkg.Types.Context, ::Array{Pkg.Types.PackageSpec,1}) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.2/Pkg/src/API.jl:167
 [6] up at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.2/Pkg/src/API.jl:143 [inlined]
 [7] do_up!(::Dict{Symbol,Any}, ::Array{Pkg.Types.PackageSpec,1}, ::Dict{Symbol,Any}) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.2/Pkg/src/REPLMode.jl:483
 [8] #invokelatest#1(::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}, ::typeof(Base.invokelatest), ::Any, ::Any, ::Vararg{Any,N} where N) at ./essentials.jl:790
 [9] invokelatest(::Any, ::Any, ::Vararg{Any,N} where N) at ./essentials.jl:789
 [10] do_cmd!(::Pkg.REPLMode.Command, ::REPL.LineEditREPL) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.2/Pkg/src/REPLMode.jl:403
 [11] #do_cmd#23(::Bool, ::typeof(Pkg.REPLMode.do_cmd), ::REPL.LineEditREPL, ::String) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.2/Pkg/src/REPLMode.jl:382
 [12] do_cmd at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.2/Pkg/src/REPLMode.jl:378 [inlined]
 [13] (::getfield(Pkg.REPLMode, Symbol("##28#31")){REPL.LineEditREPL,REPL.LineEdit.Prompt})(::REPL.LineEdit.MIState, ::Base.GenericIOBuffer{Array{UInt8,1}}, ::Bool) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.2/Pkg/src/REPLMode.jl:610
 [14] #invokelatest#1 at ./essentials.jl:790 [inlined]
 [15] invokelatest at ./essentials.jl:789 [inlined]
 [16] run_interface(::REPL.Terminals.TextTerminal, ::REPL.LineEdit.ModalInterface, ::REPL.LineEdit.MIState) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.2/REPL/src/LineEdit.jl:2306
 [17] run_frontend(::REPL.LineEditREPL, ::REPL.REPLBackendRef) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.2/REPL/src/REPL.jl:1038
 [18] run_repl(::REPL.AbstractREPL, ::Any) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.2/REPL/src/REPL.jl:201
 [19] (::getfield(Base, Symbol("##737#739")){Bool,Bool,Bool,Bool})(::Module) at ./client.jl:390
 [20] #invokelatest#1 at ./essentials.jl:790 [inlined]
 [21] invokelatest at ./essentials.jl:789 [inlined]
 [22] run_main_repl(::Bool, ::Bool, ::Bool, ::Bool, ::Bool) at ./client.jl:374
 [23] exec_options(::Base.JLOptions) at ./client.jl:312
 [24] _start() at ./client.jl:464
```

**Comment by fredrikekre** [comment]
Hmm, but in the linked manifest `SixelTerm has` a `git-tree-sha1` entry.

**Comment by 00vareladavid** [comment]
Seems like a dup of #1124

**Comment by oschulz** [comment]
Oh, yes, sure looks like it.

**Comment by fredrikekre** [comment]
That issue was for `up` though. Do we need a similar fix for the `add` code path?

**Comment by 00vareladavid** [comment]
Where do you see `add` Fredrik? The error seems to be triggered from `update` in this issue as well.

**Comment by fredrikekre** [comment]
:man_facepalming: Maybe thats why it did not reproduce when trying `add`...

Closing as fixed then, will be in 1.2 final.

**Comment by oschulz** [comment]
Tried again to shorten my Project.toml, taking all the dev-packages out to make create a good test case. Unfortunately the error goes away as soon as I remove to much. Sorry ...

So let's trust that this really is the same as #1224. :-)
